### PR TITLE
[Merged by Bors] - feat: Polynomial.mul_modByMonic

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds simp lemma for `(p * q) %ₘ q = 0` and `(q * p) %ₘ q = 0`.

Also corrects a misspelling: `dvd_iff_modByMonic_eq_zero` should be `modByMonic_eq_zero_iff_dvd`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Div.lean [hunk 1]
```diff
@@ -395,7 +395,7 @@ theorem map_modByMonic [Ring S] (f : R →+* S) (hq : Monic q) :
   (map_mod_divByMonic f hq).2
 #align polynomial.map_mod_by_monic Polynomial.map_modByMonic
 
-theorem dvd_iff_modByMonic_eq_zero (hq : Monic q) : p %ₘ q = 0 ↔ q ∣ p :=
+theorem modByMonic_eq_zero_iff_dvd (hq : Monic q) : p %ₘ q = 0 ↔ q ∣ p :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Polynomial/Div.lean:398]: Please add the deprecated alias

File: Mathlib/Data/Polynomial/Div.lean [hunk 2]
```diff
@@ -696,6 +701,11 @@ theorem eval_divByMonic_pow_rootMultiplicity_ne_zero {p : R[X]} (a : R) (hp : p
       (Nat.lt_succ_self _) (dvd_of_mul_right_eq _ this)
 #align polynomial.eval_div_by_monic_pow_root_multiplicity_ne_zero Polynomial.eval_divByMonic_pow_rootMultiplicity_ne_zero
 
+@[simp]
+lemma mul_left_modByMonic (hq : q.Monic) : (p * q) %ₘ q = 0 := by
+  rw [modByMonic_eq_zero_iff_dvd hq]
+  exact dvd_mul_left q p
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Polynomial/Div.lean:707]: I was confused why this was so far from the other one, but it looks like it needs commutativity. Would it be helpful to add documentation comments to these two lemmas pointing at each other?

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

File: Mathlib/Data/Polynomial/Div.lean (hunk 2 above)
Comment (BoltonBailey) [reply Mathlib/Data/Polynomial/Div.lean:707]: Good point!

File: Mathlib/Data/Polynomial/Div.lean [hunk 3]
```diff
@@ -410,11 +410,20 @@ theorem dvd_iff_modByMonic_eq_zero (hq : Monic q) : p %ₘ q = 0 ↔ q ∣ p :=
     rw [degree_mul' hlc, degree_eq_natDegree hq.ne_zero,
       degree_eq_natDegree (mt leadingCoeff_eq_zero.2 hrpq0)] at this
     exact not_lt_of_ge (Nat.le_add_right _ _) (WithBot.some_lt_some.1 this)⟩
-#align polynomial.dvd_iff_mod_by_monic_eq_zero Polynomial.dvd_iff_modByMonic_eq_zero
+#align polynomial.dvd_iff_mod_by_monic_eq_zero Polynomial.modByMonic_eq_zero_iff_dvd
+
+@[deprecated] alias dvd_iff_modByMonic_eq_zero := modByMonic_eq_zero_iff_dvd
```
Comment (YaelDillies) [inline Mathlib/Data/Polynomial/Div.lean:415]:
```suggestion
-- 2024-03-23
@[deprecated] alias dvd_iff_modByMonic_eq_zero := modByMonic_eq_zero_iff_dvd
```

File: Mathlib/Data/Polynomial/Div.lean [hunk 4]
```diff
@@ -410,11 +410,20 @@ theorem dvd_iff_modByMonic_eq_zero (hq : Monic q) : p %ₘ q = 0 ↔ q ∣ p :=
...
+
+/-- See `Polynomial.mul_left_modByMonic` for the other multiplication order. That version, unlike
+this one, requires commutativity. -/
+@[simp]
+lemma mul_right_modByMonic (hq : q.Monic) : (q * p) %ₘ q = 0 := by
```
Comment (YaelDillies) [inline Mathlib/Data/Polynomial/Div.lean:420]:
```suggestion
lemma self_mul_modByMonic (hq : q.Monic) : (q * p) %ₘ q = 0 := by
```

File: Mathlib/Data/Polynomial/Div.lean [hunk 5]
```diff
@@ -696,6 +705,13 @@ theorem eval_divByMonic_pow_rootMultiplicity_ne_zero {p : R[X]} (a : R) (hp : p
       (Nat.lt_succ_self _) (dvd_of_mul_right_eq _ this)
 #align polynomial.eval_div_by_monic_pow_root_multiplicity_ne_zero Polynomial.eval_divByMonic_pow_rootMultiplicity_ne_zero
 
+/-- See `Polynomial.mul_right_modByMonic` for the other multiplication order. This version, unlike
+that one, requires commutativity. -/
+@[simp]
+lemma mul_left_modByMonic (hq : q.Monic) : (p * q) %ₘ q = 0 := by
```
Comment (YaelDillies) [inline Mathlib/Data/Polynomial/Div.lean:711]:
```suggestion
lemma mul_self_modByMonic (hq : q.Monic) : (p * q) %ₘ q = 0 := by
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**